src/js/Cell.js

```diff
@@ -16,22 +17,41 @@ class Cell extends MazeEntity {
      * @returns {Array.<Cell>} List of neighboring cells
      */
     getNeighbors() {
-
+        return this.getNeighboringEntitiesList(2)
     }
 
     /**
-     * @param {Cell} cell 
+     * @param {Cell} cell
      * @returns {boolean} Represents whether there is a removed wall between cells
      */
     canAccessCell(cell){
+      return (
+          cell.visited==false
+          && this.isNeighboringCell(cell)
+          && this.getInterveningWall(cell).state === WallState.PENDING
+      )
+    }
 
+    getInterveningWall(cell){
+      if(this.isNeighboringCell(cell)){
+        return this.maze.data[(this.row+cell.row)/2][(this.col+cell.col)/2]
+      }
+      return null;
+    }
+
+    isNeighboringCell(cell){
```

Comment:
can we just use our existing `getNeighbors` method, and then check whether this cell is in the returned list? that way we can avoid duplicating logic.

Reply:
Sure, that would work too, I don't feel strongly so I can change it